examples/jsm/postprocessing/OutlinePass.js

```diff
@@ -105,7 +105,8 @@ class OutlinePass extends Pass {
 			blending: NoBlending,
 			depthTest: false,
 			depthWrite: false,
-			transparent: true
+			transparent: true,
+			premultipliedAlpha: true
```

Comment:
Setting `NoBlending` and `premultipliedAlpha: true` and `transparent: true` at the same time?

Reply:
@Mugen87 I appreciate your desire to work on this. I just think it is going to take a lot of thought to get opacity properly-supported in `EffectComposer`.

The `transparent` flag is for user convenience, in lieu of having to set the blending. I am not sure it is wise to be setting both.(Is the `transparent` flag is needed for the transparent rendering group?).

Is `premultipliedAlpha` even supported with `NoBlending`?

Reply:
>Since the shader implementation assumes premultiplied alpha

The drawing buffer has premultiplied alpha, but the output of the shader may or may not have it.

You only need to set `.premultipliedAlpha` if it is required to get the correct answer. Setting it when it is not needed just causes more confusion.

`.premultipliedAlpha` is a flag, just like `transparent, to guide the renderer in setting the correct blending function. It only applies to the built-in blending functions.

`.premultipliedAlpha`  means the output of the _shader_ has RGB multiplied by alpha. In that case, it triggers the renderer to select a blending function _different_ from the typical one.

`.premultipliedAlpha` does not apply when `NoBlending` is set.

To understand these concepts, I'd start with understanding the `setBlending()` function in `WebGLState.js`.